packages/bitcore-lib/test/script/interpreter.js

```diff
@@ -416,21 +389,22 @@ describe('Interpreter', function() {
 
           var flags = getFlags(vector[2]);
           var map = {};
-          inputs.forEach(function(input) {
+          for (let input of inputs) {
             var txid = input[0];
             var txoutnum = input[1];
             var scriptPubKeyStr = input[2];
             if (txoutnum === -1) {
               txoutnum = 0xffffffff; //bitcoind casts -1 to an unsigned int
             }
             map[txid + ':' + txoutnum] = Script.fromBitcoindString(scriptPubKeyStr);
-          });
+          };
 
           var tx = new Transaction(txhex);
           var allInputsVerified = true;
-          tx.inputs.forEach(function(txin, j) {
+          for (let j in tx.inputs) {
```

Comment:
I think this should be for of again. Although you are passing j into verify, here:
`
var verified = interp.verify(scriptSig, scriptPubkey, tx, j, flags);
`

Are you meaning to pass string representations of tx.inputs' indices?

Reply:
keeping this as a for..in because as you noted the index is used in the test case. Yeah, a string fine. `array[0] === array['0']`